mnq/sns-instances-notification-system/publisher-server/cmd/publisher/main.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"github.com/gin-gonic/gin"
+	"log"
+	"net/http"
+	"publisher-server/internal/handlers"
+	"publisher-server/internal/sns_client"
+)
+
+func main() {
+	sns_client.InitAwsSnsClient()
+
+	router := gin.Default()
+	router.LoadHTMLGlob("../../www/templates/*")
+	router.Static("/", "../../www")
+	router.POST("/low-CPU-usage", handlers.LowCPUUsageHandler)
+	router.POST("/high-CPU-usage", handlers.HighCPUUsageHandler)
+
+	router.NoRoute(func(context *gin.Context) {
+		context.HTML(http.StatusNotFound, "message.html", gin.H{
+			"Title":   "404 Not Found",
+			"Message": "Oh no 😢 The tutorial lost you",
```

Comment:
**nit(emoji):** some people's terminals don't like emojis and will crash/break somehow. It's safer not to put them in log statements.

Reply:
I agree on a terminal. Does it still apply the same way if it is on a HTML page, in a webserver that runs inside a container ?